### PR TITLE
Module4 Exercise Vagrant file and config changes

### PR DESCRIPTION
**Pull request by rohanraj14**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  # View the documentation for the provider you are using for more
+  # information on available options.
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Ansible, Chef, Docker, Puppet and Salt are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+  # config.vm.provision "shell", inline: <<-SHELL
+  #   apt-get update
+  #   apt-get install -y apache2
+  # SHELL
```
Comment (Jonesey13) [inline Vagrantfile]: Might as well remove all these unused lines
```suggestion
```

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev \
+    xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
+    # TODO: Install pyenv
+    git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+    echo 'export PYENV_ROOT="$HOME/.pyenv"' >> ~/.profile
+    echo 'export PATH="$PYENV_ROOT/bin:$PATH"' >> ~/.profile
+    echo -e 'if command -v pyenv 1>/dev/null 2>&1; then\n  eval "$(pyenv init -)"\nfi' >> ~/.profile
+    echo pyenv install 3.7.7
+    echo pyenv global 3.7.7
+    exec "$SHELL"
```
Comment (Jonesey13) [inline Vagrantfile]: This won't actually run pyenv as echo will just print what's after it (remove the echo).
```suggestion
    source ~/.profile
    pyenv install 3.7.7
    pyenv global 3.7.7
```
Also `exec "$SHELL"` stops the shell script from proceeding any further (it's intended for interactive use only not for an automated script like this. Instead we use `source` to run the commands from the previous 3 lines (that are added to `~/.profile`)

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
This is nearly there! There's one problem that prevents python/poetry from being installed by the script (see my PR comments)

**Review by hugh-emerson — APPROVED** [review]
Looks good to me - merge when you're ready 👍 

It's usually worth mentioning things like this in the README - just letting the reader know they have the option of running the app via Vagrant

File: Vagrantfile (hunk 1 above)
Comment (rohanraj14) [reply Vagrantfile]: Removed

File: Vagrantfile (hunk 2 above)
Comment (rohanraj14) [reply Vagrantfile]: Done as suggested.

**Comment by rohanraj14** [comment]
Sure will try in new commits

**Pull request merged**